trade-finance-manager-ui/server/controllers/case/cancellation/validation/validate-reason-for-cancelling.ts

```diff
@@ -0,0 +1,19 @@
+import { MAX_CHARACTER_COUNT } from '@ukef/dtfs2-common';
+import { ErrorSummaryViewModel, ReasonForCancellingErrorsViewModel } from '../../../../types/view-models';
+
+const REASON_TOO_LONG_MESSAGE = `Reason for cancelling must be ${MAX_CHARACTER_COUNT} characters or less`;
+
+/**
+ * @param reason The reason for cancelling
+ * @returns a reason for cancelling errors view model
+ */
+export const validateReasonForCancelling = (reason: string): ReasonForCancellingErrorsViewModel => {
+  const errorSummary: ErrorSummaryViewModel[] = [];
+
+  const reasonForCancellingErrorMessage = reason.length > MAX_CHARACTER_COUNT ? REASON_TOO_LONG_MESSAGE : undefined;
```

Comment:
```suggestion
  const reasonForCancellingErrorMessage = reason.length > MAX_CHARACTER_COUNT && REASON_TOO_LONG_MESSAGE;
```

Reply:
This would pass `false` into the `ReasonForCancellingErrorsViewModel`, I'm not sure how it would be handled by the nunjucks template but thats not the current pattern

Reply:
Feels very strange that we need to explicitly return an undefined value. Is this approach used in other places?

Reply:
This just allows me to have one return at the end
```
  return { errorSummary, reasonForCancellingErrorMessage };
```

rather than 

```
export const validateReasonForCancelling = (reason: string): ReasonForCancellingErrorsViewModel => {
  if (reason.length > MAX_CHARACTER_COUNT) {
    return {
      errorSummary: [{{ text: REASON_TOO_LONG_MESSAGE, href: 'reason-for-cancelling' }}],
      reasonForCancellingErrorMessage: REASON_TOO_LONG_MESSAGE,
    }
  }

  return { errorSummary: [] };
};
```

Reply:
It's based on the other typescript validation files (e.g. `trade-finance-manager-ui\server\controllers\utilisation-reports\find-reports-by-year\search-input-validator.ts`) which allows this validation to be extending on if there was more than one input to be validated.

Reply:
Thanks @MarRobSoftwire I think there are some other ways we could do it, but happy with this existing approach.